### PR TITLE
LIMS-1917 Inconsistencies related to significant digits in uncertainties

### PR DESCRIPTION
**Pull request by Espurna**
[JIRA](https://jira.bikalabs.com/browse/LIMS-1917)

### CODE REVIEW

File: bika/lims/browser/calcs.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def calculate(self, uid=None):
         mapping = {}
 
         # values to be returned to form for this UID
-        Result = {'uid': uid, 'result': form_result}
+        Result = {'uid': uid, 'result': form_result, 'result_str': self.value}
```
Comment (xispa) [inline bika/lims/browser/calcs.py]: Shouldn't be:

```python
Result = {'uid': uid, 'result': form_result, 'result_str': form_result}
```
instead?

File: bika/lims/content/analysis.py [hunk 2]
```diff
@@ -940,7 +941,8 @@ def getPrecision(self, result=None):
             uncertainty = self.getUncertainty(result)
             if uncertainty == 0:
                 return 1
-            return abs(get_significant_digits(uncertainty))
+            significant_digit = get_significant_digits(uncertainty)
+            return 0 if significant_digit > 0 else abs(significant_digit)
```
Comment (xispa) [inline bika/lims/content/analysis.py]: Better to direclty add this logic inside [`get_significant_digits`](https://github.com/bikalabs/bika.lims/blob/a66dd805ab50196030c7c1c7341f98371dc113bd/bika/lims/utils/analysis.py#L36]) function.

File: bika/lims/content/analysisservice.py [hunk 3]
```diff
@@ -1240,7 +1242,8 @@ def getPrecision(self, result=None):
             # https://jira.bikalabs.com/browse/LIMS-1334
             if uncertainty == 0:
                 return 1
-            return abs(get_significant_digits(uncertainty))
+            significant_digit = get_significant_digits(uncertainty)
+            return 0 if significant_digit > 0 else abs(significant_digit)
```
Comment (xispa) [inline bika/lims/content/analysisservice.py]: See comment https://github.com/bikalabs/bika.lims/pull/1717/files#r46287608

File: bika/lims/content/referenceanalysis.py [hunk 4]
```diff
@@ -368,7 +368,8 @@ def getPrecision(self, result=None):
             uncertainty = self.getUncertainty(result)
             if uncertainty == 0:
                 return 1
-            return abs(get_significant_digits(uncertainty))
+            significant_digit = get_significant_digits(uncertainty)
+            return 0 if significant_digit > 0 else abs(significant_digit)
```
Comment (xispa) [inline bika/lims/content/referenceanalysis.py]: See comment https://github.com/bikalabs/bika.lims/pull/1717/files#r46287608

**Comment by Espurna** [comment]
pau@mendel:~/dev/Espurna/bikalims/zinstance$ bin/test -v -p -s bika.lims --suite-name='bika.lims.tests' --test='test_get_significant_digits'
Test-module import failures:

Module: bika.lims.tests.test_doctests

TypeError: Module bika.lims.tests.test_doctests does not define any tests


Module: bika.lims.tests.test_robot

TypeError: Module bika.lims.tests.test_robot does not define any tests


Running tests at level 1
Running bika.lims.testing.BikaTestingLayer:Functional tests:
  Set up plone.testing.zca.LayerCleanup in 0.000 seconds.
  Set up plone.testing.z2.Startup in 0.149 seconds.
  Set up plone.app.testing.layers.PloneFixture in 12.253 seconds.
  Set up bika.lims.testing.BikaTestLayer in 51.093 seconds.
  Set up bika.lims.testing.BikaTestingLayer:Functional in 0.000 seconds.
  Running:
                                                                               
  Ran 1 tests with 0 failures and 0 errors in 0.013 seconds.
Tearing down left over layers:
  Tear down bika.lims.testing.BikaTestingLayer:Functional in 0.000 seconds.
  Tear down bika.lims.testing.BikaTestLayer in 0.011 seconds.
  Tear down plone.app.testing.layers.PloneFixture in 0.052 seconds.
  Tear down plone.testing.z2.Startup in 0.006 seconds.
  Tear down plone.testing.zca.LayerCleanup in 0.006 seconds.

Test-modules with import problems:
  bika.lims.tests.test_doctests
  bika.lims.tests.test_robot
pau@mendel:~/dev/Espurna/bikalims/zinstance$

File: bika/lims/browser/js/bika.lims.utils.calcs.js [hunk 5]
```diff
@@ -195,8 +195,13 @@ function CalculationUtils() {
...
                         result = $(data['results'])[i];
-
-                        $("input[uid='"+result.uid+"']").filter("input[field='Result']").val(result.result);
+                        // We have to get the decimals because if they are .0, the will be ignored
+                        var decimals = 0;
+                        if (result.result_str.indexOf(".") > -1){
+                            decimals = result.result_str.split('.')[1].length;
+                        }
+                        var result_with_decimals = parseFloat(result.result_str).toFixed(decimals)
+                        $("input[uid='"+result.uid+"']").filter("input[field='Result']").val(result_with_decimals);
```
Comment (rockfruit) [inline bika/lims/browser/js/bika.lims.utils.calcs.js]: this change causes calculations to fail.

**Comment by xispa** [comment]
@rockfruit PR #1767 fixes the issues rised from here

**Pull request merged**